Ui/Component/Listing/Column/Data.php

```diff
@@ -27,9 +27,18 @@ public function prepareDataSource(array $dataSource)
             if (is_array($data)) {
                 foreach ($data as $key => $value) {
                     if (is_array($value)) {
-                        $value = implode(', ', $value);
+                        $stringValue = '';
+                        foreach ($value as $k => $v) {
```

Comment:
Great idea! I'm thinking we should split this formatter out into a separate function - and call it recursively for nested values (up to a certain depth limit).